Source/Core/Core/NetPlayServer.cpp

```diff
@@ -867,6 +870,46 @@ unsigned int NetPlayServer::OnData(sf::Packet& packet, Client& player)
       return 1;
     }
   }
+  break;
+
+    case NP_MSG_SYNC_GECKO_CODES:
+  {
+    // Receive Status of Gecko Sync
+    MessageId sub_id;
+    packet >> sub_id;
+
+    // Check If Sync was successful or not
+    switch (sub_id)
+    {
+    case SYNC_GECKO_CODES_SUCCESS:
+    {
+      if (m_start_pending)
+      {
+        m_gecko_codes_synced_players++;
+        if (m_gecko_codes_synced_players >= m_players.size() - 1)
+        {
+          m_dialog->AppendChat(GetStringT("All players synchronized."));
+          StartGame();
```

Comment:
This definitely cannot be handled this way, as save sync also calls this. If both syncs are going to call game launch, we need a couple booleans and another function to check that both are true before launching.